processmanager/processinfo.go

```diff
@@ -591,20 +591,12 @@ func (pm *ProcessManager) SynchronizeProcess(pr process.Process) {
 		return
 	}
 	if len(mappings) == 0 {
-		// Valid process without any (executable) mappings. All cases are
-		// handled as process exit. Possible causes and reasoning:
-		// 1. It is a kernel worker process. The eBPF does not send events from these,
-		//    but we can see kernel threads here during startup when tracer walks
-		//    /proc and tries to synchronize all PIDs it sees.
-		//    The PID should not exist anywhere, but we can still double check and
-		//    make sure the PID is not tracked.
-		// 2. It is a normal process executing, but we just sampled it when the kernel
-		//    execve() is rebuilding the mappings and nothing is currently mapped.
-		//    In this case we can handle it as process exit because everything about
-		//    the process is changing: all mappings, comm, etc. If execve fails, we
-		//    reaped it early. If execve succeeds, we will get new synchronization
-		//    request soon, and handle it as a new process event.
-		pm.processPIDExit(pid)
+		// TODO: Check if main thread has exited, e.g. /proc/PID/stat will show zombie
```

Comment:
Zombie state is not the "main thread exited" state. Zombie means that the process has exited completed and released all resources, but its still in process table waiting for a parent process to call `waitpid` to reap the exit status.

Reply:
Yes, we can use the zombie test in this case as the condition we'd be testing for would be: len(mappings) == 0 AND zombie state which should be a good indicator for main thread exit. It also fails safe, e.g. if we hit a race with process exit, at worst we delay process cleanup for a few seconds (5 in this PR).

There could also be other tests we can use, this was the first one that sprang to mind. I also checked if `/proc/PID/task/PID` goes away when main thread calls `pthread_exit` but no.